src/openms/include/OpenMS/SYSTEM/File.h

```diff
@@ -183,13 +182,53 @@ namespace OpenMS
     */
     static String findExecutable(const String& toolName);
 
+    /**
+      @brief Obtain a temporary filename, ensuring automatic deletion upon exit
+
+      The file is not actually created and only deleted at exit if it exists.
+      
+      However, if 'alternative_file' is given and not empty, no temporary filename
+      is created and 'alternative_file' is returned (and not destroyed upon exit).
+      This is useful if you have an optional
+      output file, which may, or may not be requested, but you need its content regardless,
+      e.g. for intermediate plotting with R.
+      Thus you can just call this function to get a file which can be used and gets automatically
+      destroyed if needed.
+
+      @param alternative_file If this string is not empty, no action is taken and it is used as return value
+      @return Full path to a temporary file
+    */
+    static const String& getTemporaryFile(const String& alternative_file = "");
```

Comment:
some suggestion:
- make TemporaryFile a class that has a destructor that removes the file. That way the user can control if he wants to delete the file in a local context or on program exit.
- the behavior of returning the alternative file name is - in my view -  trying to be smart but is imho less transparent than just checking for empty string at the actual site of useage